cmd_exec.c

```diff
@@ -128,3 +121,40 @@ int cmd_exec(context_t *context) {
   free(command);
   return ret;
 }
+
+int cmd_exec_sync(char **command) {
+  pid_t child = fork();
+
+  if (child == 0) { /* child */
+    execvp(command[0], command);
+
+    /* if we get here, there was an error */
+    perror("execvp failed");
+    exit(errno);
+  } else { /* parent */
+    int status = 0;
+    waitpid(child, &status, 0);
+    return WEXITSTATUS(status);
+  }
+}
+
+void cmd_exec_detached(char **command) {
+  pid_t child = fork();
+
+  if (child == 0) { /* child */
+    pid_t grandchild = fork();
+    if (grandchild == 0) { /* grandchild */
```

Comment:
Is double-fork preferred for solving this? I wonder if adding a SIGCHLD handler to xdotool would solve this just as well.

If double-fork is preferred (where the 1st child does the reaping/waidpid), then a comment describing why this mechanism was chosen might be useful for future readers (ourselves included).

Thoughts?